--- http-live-streaming-tools/Makefile	2010-08-30 10:27:02.367948681 +0300
+++ http-live-streaming-tools/Makefile	2010-08-30 10:29:49.374933976 +0300
@@ -1,5 +1,13 @@
-all:
-	gcc -Wall -g live_segmenter.c -o live_segmenter -lavformat -lavcodec -lavutil -lbz2 -lm -lz -lfaac -lmp3lame -lx264 -lfaad -lpthread
+CC := gcc
+CFLAGS := -g
+
+all: live_segmenter
+
+%.o: %.c
+	$(CC) -Wall $(CFLAGS) live_segmenter.c -c
+
+live_segmenter: live_segmenter.o
+	$(CC) -Wall $(CFLAGS) $< -o $@ $(LDFLAGS) -lavformat -lavcodec -lavutil -lbz2 -lm -lz -lfaac -lmp3lame -lx264 -lfaad -lpthread
 
 clean:
 	rm -f live_segmenter
